scss/_input-group.scss

```diff
@@ -169,17 +168,11 @@
   }
 
   // Negative margin to only have a single, shared border between the two
-  &:first-child {
-    > .btn,
-    > .btn-group {
-      margin-right: (-$input-btn-border-width);
-    }
-  }
-  &:last-child {
+  &:not(:first-child) {
     > .btn,
     > .btn-group {
       z-index: 2;
-      margin-left: (-$input-btn-border-width);
+      margin-left: (-$input-btn-border-width) * 2;
```

Comment:
Kind of a see-sawing situation, because the original `.input-group-btn` has double borders in your rectangle, while `.input-group-addon` does not.